file/writable_file_writer.cc

```diff
@@ -155,6 +155,11 @@ IOStatus WritableFileWriter::Close() {
   writable_file_.reset();
   TEST_KILL_RANDOM("WritableFileWriter::Close:1", rocksdb_kill_odds);
 
+  if (checksum_generator_ != nullptr && !checksum_finalized_) {
```

Comment:
If `!s.ok()`, we do not have to call `Finalize()`.

Reply:
@riversand963 Will make the update